main.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const PROMPT = "> "
+const ERROR = "ERROR"
+
+func main() {
+	scanner := bufio.NewScanner(os.Stdin)
+
+	for {
+		fmt.Print(PROMPT)
+
+		scanner.Scan()
+
+		raw_text := scanner.Text()
+
+		text := removeDuplicateSpaces(raw_text)
+
+		tokens := strings.Split(text, " ")
+
+		result, err := process(tokens)
+
+		if err != nil {
+			fmt.Println(err)
+		} else {
+			fmt.Println(text, "=", result)
+		}
+	}
+}
+
+func process(tokens []string) (result string, err error) {
+	tokens, err = calculate(tokens, "*", "/")
+	if err != nil {
+		return
+	}
+
+	tokens, err = calculate(tokens, "+", "-")
+	if err != nil {
+		return
+	}
+
+	if len(tokens) == 1 {
+		result = tokens[0]
+	} else {
+		err = errors.New(ERROR)
+	}
+
+	return
+}
+
+func calculate(tmp_tokens []string, operator_1, operator_2 string) (tokens []string, err error) {
+	tokens = tmp_tokens
+
+	for {
+		index := -1
+		index, err = find_valid_operator(tokens, operator_1, operator_2)
+		if err != nil {
+			return
+		}
+
+		if index == -1 {
+			break
+		}
+
+		var result float64
+		result, err = evaluate(tokens[index-1 : index+2])
+		if err != nil {
+			return
+		}
+
+		tokens[index+1] = fmt.Sprintf("%g", result)
+		tokens = append(tokens[:index-1], tokens[index+1:]...)
+	}
+	return
+}
+
+func evaluate(formula []string) (result float64, err error) {
+
+	left_operand, err_1 := strconv.ParseFloat(formula[0], 10)
```

Comment:
accessing slice without checking it's length might cause runtime error (`index out of range`)
So better passing in leftOp, rightOp and Op instead of a slice